src/Ticket.php

```diff
@@ -4379,7 +4379,13 @@ public function showForm($ID, array $options = [])
                         break;
                 }
                 // Check category / entity validity
-                if (!in_array($cat->fields['entities_id'], getSonsOf('glpi_entities', $options['entities_id']))) {
+                if (
+                    in_array($cat->fields['entities_id'], getSonsOf('glpi_entities', $options['entities_id']))
+                    || (
+                        $cat->isRecursive()
+                        && !in_array($cat->fields['entities_id'], getAncestorsOf('glpi_entities', $options['entities_id']))
+                    )
+                ) {
```

Comment:
If I understand correctly, we want to drop a category whenever it is not available on the entity passed in `$options['entities_id']`. If I am correct it means that we must refuse the category
if the category IS NOT recursive and its entity does not match `$options['entities_id']`
OR
if the category IS and `$options['entities_id']` is not part of the sons of the entities of the category 

```suggestion
                if (
                    (
                        !$cat->isRecursive()
                        && $cat->fields['entities_id'] !== $options['entities_id']
                    )
                    || (
                        $cat->isRecursive()
                        && !in_array($options['entities_id'], getSonsOf('glpi_entities', $cat->fields['entities_id']))
                    )
                ) {
```